docs/concepts/case-study-openbazaar.md

```diff
@@ -0,0 +1,108 @@
+- With Filecoin, the ability to incentivize nodes for long-term storage
+- A robust enough solution to enable the team to focus on product development, not infrastructure development
+- Support from and ongoing collaboration with the IPFS core team
+
+## How OpenBazaar uses IPFS
+
+OpenBazaar's implementation is built upon a fork of the [`go-ipfs`](https://github.com/ipfs/go-ipfs) reference implementation, customized for their specific needs:
+
+- Persistent storage of at least a week (to remember buyers' shopping carts and allow for merchants who may not be online 24/7)
+- Support for Bitcoin wallets
+- A customized search engine that crawls the IPFS network to update shop content (the crawler maintains the inventory for OpenBazaar's and Haven's front end)
+- Circuit relay capabilities for working around firewalls and mobile iOS networks
+- The use of `go-ipfs` without IPNS, to optimize for OpenBazaar's extremely large number of product images
+- The use of [`libp2p`](https://libp2p.io/), but with its original addressing scheme
+
+### The architecture
+
+At its highest level, the OpenBazaar network is made out of many nodes, whether desktop/laptop (OpenBazaar) or mobile (Haven). Each node is either a merchant or a seller, with a few exceptions; OB1 does run some gateway nodes and caching nodes themselves. Every node has either the OpenBazaar or Haven app installed, and "inside" the respective app is a complete `go-ipfs` node.
+
+<img src="./images/case-studies/img-openbazaar-nodes.png" alt="OpenBazaar high-level architecture">
+
+### Customizing &amp; extending
+
+As mentioned above, OB1 implemented a variety of customizations and extensions to get the most out of their use of IPFS. One extension is an asynchronous messaging layer. By building this additional storage layer with IPFS, the OB1 team created a system in which messages could be retained when users are offline, and sent or received later.
+
+Another use-case-specific customization was a customized search crawler to comb the network for new content. As merchants add new merchandise, the network needs to display that to the buyers in a timely fashion. OB1's custom crawler searches the network for new content and keeps track of new items in the system so buyers can see and purchase the latest merchandise.
+
+Additionally, OB1 created a TOR router for IPFS, as well as a circuit-relay workaround for firewalls for Haven users on iOS. The ability to create these types of modular add-ins represents a valuable flexibility inherent in IPFS.
+
+## OpenBazaar + IPFS: the future
+
+As OpenBazaar and Haven continue to grow, OB1 is looking at potentially integrating serverless, distributed peer-to-peer database OrbitDB in its IPFS implementation. OpenBazaar and Haven currently use flat files in a directory, which is a bit unusual for a marketplace. Building an e-commerce network on top of flat files isn’t ideal, so an OrbitDB implementation may be a near-term boost.
```

Comment:
s/integrating serverless/integrating the serverless/

Reply:
👍